Jonathan Dorety

jdorety opened the pull request:
@lndubose 

MVP functionality for Day 1, still some styling work needed

jdorety committed:
Basic project setup

jdorety committed:
Add Bootstrap libraries

jdorety committed:
Add SearchBar

jdorety committed:
Initial header bar

Still needs final styling

jdorety committed:
Add user posts (images)

jdorety committed:
Display CommentSection in PostContainer

jdorety committed:
Complete CommentSection Day 1 MVP functionality

jdorety committed:
Improve styling for SearchBar

jdorety committed:
Move Comments input below TimePosted

jdorety committed:
Move SearchBar down slightly

jdorety committed:
Add PropTypes check

jdorety committed:
Add PropTypes check for TimePosted

jdorety committed:
Add PropTypes check for CommentSection

jdorety committed:
Remove SearchInput component

Was redundant, and did not need to exist as separate component from 
SearchBar

jdorety committed:
Change Comments field key mapping

Changing key ID's from index position to comment text value

jdorety committed:
Styling Comments

jdorety committed:
Correct comment display structuring

jdorety committed:
Comments section component styled

Need to write timestamp functionality and style, and style comment input 
box

jdorety committed:
Add Moments.js

jdorety committed:
Add correct Moment libraries

jdorety committed:
Work on TimePosted

jdorety committed:
TimePosted fromNow functionality implemented

jdorety committed:
Complete basic styling on Comments area

jdorety committed:
Set state in App.js with componentDidMount()

jdorety committed:
Tweak TimeStamp color

jdorety committed:
Refactor CommentSection to class, implement componentDidMount

jdorety committed:
Moving TimePosted and Input components into CommentSection

jdorety committed:
Comment adding functionality created for CommentSection.js

jdorety committed:
Add PropTypes additions to CommentSection

jdorety committed:
Make TimePosted component render uppercase

jdorety committed:
SearchBar styling

jdorety committed:
Add Likes functionality works with the heart icons

jdorety committed:
Comment input styling

jdorety committed:
Basic search bar functionality implemented

Seems to be one step behind

jdorety committed:
Make search filter snappier

jdorety committed:
Add comments

kyullog commented:
Day 2 MVP complete!

jdorety committed:
Mobile styling

jdorety committed:
Change cursor style on like buttons to pointer

jdorety committed:
Refactor addLikeHandler

Lndubose commented:
# Day 1
## Tasks (Day I)
- [x] There are three major container components that you'll need to implement for this project: the Search Bar, the Post Container, and the Comment Section.
- [x] At the end of Day I there will be a single instance of the Search Bar being rendered at the top of the page, as well as a Post Container and a Comment Section for every piece of mock data in the dummy-data.js file.
- [x] The root App component of your application should import the dummy data from the dummy-data.js file with import dummyData from './dummy-data'; and iterate over said data, passing each individual object as a prop to an instance of PostContainer.
- [x] Each PostContainer component will then pass the array of comments on each post object as a prop to an instance of the CommentSection component.
- [x] The CommentSection component will receive the array of comments as props and render a Comment component with the username of the poster as well as the post's text. 
- [x] Additionally, there should be an input box that allows users to submit a new comment for any post. We'll work on posting new comments tomorrow.
 - [x] Be sure to check the Types of the data you are passing around as props in the components that will be using props to present data as DOM elements. This should be linked to your Comment component that Comment Section will render and potentially to your Post component that Post Container will render.
- [x] You are free to leverage the Bootstrap library for this project for the purposes of theming and styling. I recommend the awesome reactstrap library, which is a library of Bootstrap components that have been implemented using React, so they're really easy to just drop straight into React projects.
- [x] In addition to Bootstrap for theming, you'll want to add your own styles via CSS. To keep things organized, have the CSS file that corresponds with a component live in the same directory as the component file.

## Stretch Problems (Day I)
- [ ] Implement the ability to comment on a post with the Add a comment... input.
- [x] Use the moment.js library dynamically format the timestamp into a human-readable format like how it is being displayed in the screenshot.

# Day 2
## Tasks (Day II)
### Use lifecycle methods
  - [x] In `app.js` use `componentDidMount()` to set your data to the component's state. Use the state object to pass data to your `PostContainer` component.
### Adding comments
  - [x] Lets divide up the data that we've been working with this far by separating the comments array onto a new component's state. Pass down the comments through each post to the CommentSection component. Be sure that you set the incoming `comments` props data on the state of the CommentSection component.
  - [x] Create a function in `CommentSection.js` called `addNewComment` that takes in an event and an index number. The function will add the comment that is on the event object to the post that is at that index number.
  - [x] Pass the `addNewComment` function down the component tree to where ever you have your 'Add a comment...' input.
  - [x] The 'Add a comment...' input should be wrapped in a `<form></form>` element. Using that form's `onSubmit` event handler, invoke the `addNewComment` function and pass it the required arguments it needs to add a new comment.
  - [x] Update your state with the new comment (This should trigger your component tree to "re-render" with your new comment on that post).
### Liking posts
  - [x] This will be set up a lot like the 'Add a comment...' input. Pass a function down to where your heart icon is, and use `onClick` event handler to increment that post's likes.
### Search
  - [x] Set up the search bar will like the comment input and the like button. In your function, filter out any post whose username doesn't match the search term passed in, then update the state with the resulting data.

## Stretch Problems (Day II)

- [ ] Persist your data using `localStorage`. If done correctly, you will be able to refresh your page and still see your new comments.
- [ ] Add the ability to delete a comment from your data. If your data is in `localhost`, make sure to delete it from there as well.
- [ ] Implement a third-party library that does "fuzzy" searches into your search bar functionality (ie - search terms that aren't exact, like "phils" or "coffeephilz", would still return the username "philzcoffe").

## Commits
-The commits look awesome! Very good work.

## Improvements
- Only improvement I saw was that you have setState for your searchInput function have prevState, but you do not need prevState. Therefore you can shorten it down to be `this.setState(posts: resultsArray)`

- Also make sure you have PropTypes everywhere you are passing props. You did get a lot of it but you missed some spots :)

## Rating
3

Very well done. Everything looks like it is working great and you used PropTypes almost everywhere. Also I really like how you did the componentDidMount instead of directly manipulating the state. Nice work!

jdorety committed:
Refactor search filter based on feedback

jdorety committed:
Create PostsPage component and refactor App.js accordingly

jdorety committed:
Implement Authentication component within App.js

jdorety committed:
Create LoginPage component

Give LoginPage access to Authenticate state with ability to change 
loginInfo values

jdorety committed:
Basic login functionality implemented

Still needs localstorage capability, and some bugs need to be worked out 
re: form submission.  Need to style login page, and do a sweep of all 
components to correctly implement PropTypes for all components.

jdorety committed:
Add basic localStorage functionality, and logout function w/ icon

jdorety committed:
Refine login and logout functionality

jdorety committed:
Allow logged in user name to be used for comments

jdorety committed:
yarn add styled-components

jdorety committed:
Implement PropTypes for Login

jdorety committed:
Implement more PropTypes

jdorety committed:
Fix password input type

jdorety committed:
Basic styled component implementation for Login component

jdorety committed:
Implement styled components for Search Bar

jdorety committed:
Fix proptypes

jdorety committed:
Implement single change handler for login window, refactor state shape

jdorety committed:
Slight refactoring

jdorety committed:
Fixed Search bug

jdorety committed:
Create and implement styled components for login and header

jdorety committed:
Implement styled components in comments

Icons and timestamps still styled in CSS

jdorety committed:
Implement styled components and delete App.css
